skops/hub_utils/tests/test_hf_hub.py

```diff
@@ -416,57 +450,63 @@ def test_inference(
 ):
     # test inference backend for classifier and regressor models. This test can
     # take a lot of time and be flaky.
-    client = HfApi()
-
-    repo_path = repo_path_for_inference
-    model = model_func()
-    model_path = repo_path / "model.pickle"
-
-    with open(model_path, "wb") as f:
-        pickle.dump(model, f)
-
-    version = metadata.version("scikit-learn")
-    init(
-        model=model_path,
-        requirements=[f'scikit-learn="{version}"'],
-        dst=destination_path,
-        task=task,
-        data=data.data,
-    )
+    config_path, file_format = config_json
+    if file_format != "pickle":
+        return
+    else:
+        client = HfApi()
+        repo_path = repo_path_for_inference
+        model_file = CONFIG[file_format]["sklearn"]["model"]["file"]
+        model = model_func()
+        model_path = repo_path / model_file
+
+        with open(model_path, "wb") as f:
+            pickle.dump(model, f)
 
-    # a model card is needed for inference engine to work.
-    model_card = card.Card(
-        model, metadata=card.metadata_from_config(Path(destination_path))
-    )
-    model_card.save(Path(destination_path) / "README.md")
-
-    user = client.whoami(token=HF_HUB_TOKEN)["name"]
-    repo_id = f"{user}/test-{uuid4()}"
+        version = metadata.version("scikit-learn")
+        init(
+            model=model_path,
+            requirements=[f'scikit-learn="{version}"'],
+            dst=destination_path,
+            task=task,
+            data=data.data,
+        )
 
-    push(
-        repo_id=repo_id,
-        source=destination_path,
-        token=HF_HUB_TOKEN,
-        commit_message="test message",
-        create_remote=True,
-        # api-inference doesn't support private repos for community projects.
-        private=False,
-    )
+        # a model card is needed for inference engine to work.
+        model_card = card.Card(
+            model, metadata=card.metadata_from_config(Path(destination_path))
+        )
+        model_card.save(Path(destination_path) / "README.md")
```

Comment:
add a comment to remove when https://github.com/skops-dev/skops/pull/207 is merged.